skbio/metadata/_interval.py

```diff
@@ -0,0 +1,465 @@
+                               'Interval object.')
+        if not isinstance(value, dict):
+            raise TypeError("metadata must be a dict, not %r" % value)
+        self._metadata = value
+
+    @property
+    @experimental(as_of='0.4.2-dev')
+    def dropped(self):
+        return self._interval_metadata is None
+
+
+class IntervalMetadata():
+    """ Stores Interval objects """
+    def __init__(self):
+        # stores metadata for each feature
+        self._metadata = []
+        self._intervals = IntervalTree()
+        self._is_stale_tree = False
+
+    def _reverse(self, length):
+        """ Reverse complements IntervalMetadata object.
+
+        This operation reverses all of the interval coordinates.
+        For instance, this can be used to compare coordinates
+        in the forward strand to coordinates in the reversal strand.
+
+        Parameters
+        ----------
+        length : int
+            Largest end coordinate to perform reverse complement.
+            This typically corresponds to the length of sequence.
+        """
+        for f in self._metadata:
+            # staled doesn't need to be called, since the setter for
+            # Interval will take care of this
+            invs = list(map(lambda x: (length-x[1], length-x[0]),
+                            f.intervals))
+            f.intervals = invs
+
+    def add(self, intervals, boundaries=None, metadata=None):
+        """ Adds a feature to the metadata object.
+
+        Parameters
+        ----------
+        intervals : iterable of tuple of ints
+            A list of intervals associated with the Interval object.
+        boundaries : iterable of tuple of bool
+            A list of boundaries associated with the Interval object.
+        metadata : dict
+            A dictionary of key word attributes associated with the
+            Interval object.
+
+        Examples
+        --------
+        >>> from skbio.metadata import IntervalMetadata
+        >>> interval_metadata = IntervalMetadata()
+        >>> interval_metadata.add(intervals=[(0, 2)],
+        ...                       boundaries=None, metadata={'name': 'sagA'})
+        >>> interval_metadata.add(intervals=[(40, 70)],
+        ...                       boundaries=None, metadata={'name': 'sagB'})
+        >>> list(interval_metadata.query(intervals=[(1, 2)]))
+        [<Interval: start=0, end=2, 1 contiguous interval, 1 metadata keys, \
+dropped=False>]
+
+        """
+        # Add an interval to the tree. Note that the add functionality is
+        # built within the Interval constructor.
+        Interval(interval_metadata=self,
+                 intervals=intervals,
+                 boundaries=boundaries,
+                 metadata=metadata)
+
+    def _rebuild_tree(self, intervals):
+        """ Rebuilds the IntervalTree when the tree is stale."""
+        self._intervals = IntervalTree()
+        for f in intervals:
```

Comment:
Could you make this variable name ``inv`` for consistency/readability? I think ``f`` is left over from when this was ``features``, and you're using ``inv`` in other places.